pkg/server/endpoints/admin.go

```
@@ -337,16 +340,18 @@ func (h *AdminAPIHandlers) findTrustDomainByName(ctx context.Context, trustDomai
 	return td, nil
 }
 
-func (h *AdminAPIHandlers) lookupTrustDomain(ctx context.Context, trustDomainID uuid.UUID, code int) (*entity.TrustDomain, error) {
-	td, err := h.Datastore.FindTrustDomainByID(ctx, trustDomainID)
+func (h *AdminAPIHandlers) lookupTrustDomain(ctx context.Context, trustDomainName api.TrustDomainName, code int) (*entity.TrustDomain, error) {
+	tdName := spiffeid.RequireTrustDomainFromString(trustDomainName)
```

don't use `RequireTrustDomainFromString`, it's a method for testing, it panics instead of returning an error.